repos/config/detail/lock_repos.py

```diff
@@ -49,14 +56,53 @@ def main():
         )
         yaml.dump(repos, lock_file, default_flow_style=False, allow_unicode=True)
 
+def add_attributes(dictionary, additional_attributes):
+    for k,v in additional_attributes.items():
+        dictionary[k] = v
+
+def fetch_archive_details(url, tag):
+    cwd = os.getcwd()
+    with tempfile.TemporaryDirectory() as tempdir:
+        url = url.rsplit(".git", 1)[0]
+        project = url.split('/')[-1]
+        url += f"/archive/refs/tags/{tag}.tar.gz"
+        result = subprocess.run(
+            ["curl", "-L", "-f", "-o", "archive.tar.gz", url],
+            stdout=subprocess.PIPE,
+            encoding='utf8'
+        )
+        if result.returncode != 0:
+            raise ValueError(f"Error loading {url}, {tag}: " + (result.stderr or ""))
+
+        archive_bytes = Path('archive.tar.gz').read_bytes()
+        hash = hashlib.sha256(archive_bytes).hexdigest()
+        strip_prefix = extract_archive_root_folder("archive.tar.gz", url)
+
+    return {
+        "hash":hash,
+        "url": url,
+        "strip_prefix": strip_prefix,
+        "type": "http_archive",
+    }
```

Comment:
@kilian-funk just a suggestion, but this could probably be a class and added to the function signature to make it clear what this function returns

Reply:
Yes, could be improved. I will leave it for now